packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -196,7 +196,7 @@ class AndroidApk extends ApplicationPackage {
           if (!(node is xml.XmlElement)) {
             continue;
           }
-          final xml.XmlElement xmlElement = node;
+          final xml.XmlElement xmlElement = node as xml.XmlElement;
```

Comment:
replace the '!(...is...)` with `is!` and this should just work. Or wrap this in the `is` instead of bailing early. We shouldn't need both, either way.

Reply:
Mentioned below: `is!` does not promote

Reply:
well we should still use `is!` instead of `!(...is...)` in any case. :-)